File: cpp/ecbenchmark/src/CustomFunction.cpp
#include "ecbenchmark/CustomFunction.h"

namespace ecb {

    CustomFunction::CustomFunction(Function* f, const std::string& name,
            const std::string& description, int dimensions, scalar min_domain,
            scalar max_domain, bool minimization)
    : AbstractFunction(name, description, dimensions, min_domain, max_domain, minimization),
    _function(f) {

    }

    CustomFunction::~CustomFunction() {
        if (_function) delete _function;
    }

    void CustomFunction::setFunction(Function* f) {
        this->_function = f;
    }

    Function* CustomFunction::function() const {
        return this->_function;
    }

    scalar CustomFunction::f(const std::vector<scalar>& x) {
        return function()->f(x);
    }

}

File: java/ecbenchmark/src/ecbenchmark/CustomFunction.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark;

/**
 *
 * @author jcrada
 */
public class CustomFunction extends AbstractFunction {

    private Function function;

    public CustomFunction() {
        super();
    }

    public CustomFunction(String name,
            String description, int dimensions, double min_domain,
            double max_domain, boolean minimization) {
        super(name, description, dimensions, min_domain, max_domain, minimization);
    }

    public Function getFunction() {
        return function;
    }

    public void setFunction(Function function) {
        this.function = function;
    }

    @Override
    public double f(double[] x) {
        return function.f(x);
    }
    
    
    
}
